om/hash.go

```diff
@@ -146,6 +148,61 @@ func (r *HashRepository[T]) CreateIndex(ctx context.Context, cmdFn func(schema F
 	return r.client.Do(ctx, cmdFn(r.client.B().FtCreate().Index(r.idx).OnHash().Prefix(1).Prefix(r.prefix+":").Schema())).Error()
 }
 
+// CreateAndAliasIndex creates a new index, aliases it, and drops the old index if needed.
+func (r *HashRepository[T]) CreateAndAliasIndex(ctx context.Context, cmdFn func(schema FtCreateSchema) rueidis.Completed) error {
+	alias := r.idx + "_alias"
+
+	aliasExists := true
+	if err := r.client.Do(ctx, r.client.B().FtInfo().Index(alias).Build()).Error(); err != nil {
+		aliasExists = false
+	}
+
+	var currentIndex string
+	if aliasExists {
+		infoCmd := r.client.B().FtInfo().Index(alias).Build()
+		infoResp, err := r.client.Do(ctx, infoCmd).ToMap()
+		if err != nil {
+			return err
+		}
+		message := infoResp["index_name"]
+		currentIndex = message.String()
+	}
+
+	newIndex := r.idx + "_v1"
+	if aliasExists {
+		parts := strings.Split(currentIndex, "_v")
+		if len(parts) == 2 {
+			if version, err := strconv.Atoi(parts[1]); err == nil {
+				newIndex = fmt.Sprintf("%s_v%d", r.idx, version+1)
+			}
+		}
+	}
+
+	if err := r.client.Do(ctx, cmdFn(r.client.B().FtCreate().Index(newIndex).OnHash().Prefix(1).Prefix(r.prefix+":").Schema())).Error(); err != nil {
+		return err
+	}
+
+	if aliasExists {
+		if err := r.client.Do(ctx, r.client.B().FtAliasupdate().Alias(alias).Index(newIndex).Build()).Error(); err != nil {
+			return err
+		}
+	} else {
+		if err := r.client.Do(ctx, r.client.B().FtAliasadd().Alias(alias).Index(newIndex).Build()).Error(); err != nil {
+			return err
+		}
+	}
+
+	if aliasExists && currentIndex != "" {
+		if err := r.client.Do(ctx, r.client.B().FtDropindex().Index(currentIndex).Build()).Error(); err != nil {
+			return err
+		}
+	}
+
+	r.idx = newIndex
```

Comment:
We should not change `r.idx` here. Actually, `r.idx` should be the alias. That is the point of an alias, isn't it?

Same as the above:
```go
alias := r.idx + "_alias"
```
should be just:
```go
alias := r.idx
```